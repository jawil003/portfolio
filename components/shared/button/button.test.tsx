import { render } from "@testing-library/react";
import { Button, ButtonProps } from ".";

const variants: ButtonProps[] = [
  {
    variant: "success",
    text: "Hello World",
  },
  {
    variant: "error",
    text: "Hello World",
  },
];

describe("Button", () => {
  test.each(variants)(
    "Create Button with %p",
    (variant) => {
      const testButton = render(
        // eslint-disable-next-line react/jsx-props-no-spreading
        <Button {...variant} />,
      );

      expect(
        testButton.container.firstChild,
      ).toMatchSnapshot();
    },
  );
});
