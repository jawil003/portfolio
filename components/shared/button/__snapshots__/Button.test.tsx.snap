// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Create Button with {"text": "Hello World", "variant": "error"} 1`] = `
.emotion-0 {
  background-color: #B74A66;
  padding: 0.8333333333333334em 1.6666666666666667em;
  border: none;
  outline: none;
  height: 2.8333333333333335em;
  min-width: 7.5em;
  border-radius: 1.25em;
  font-style: normal;
  font-weight: 400;
  font-weight: normal;
  font-size: 0.8333333333333334em;
  color: #fff;
}

<button
  class="emotion-0"
  type="button"
>
  Hello World
</button>
`;

exports[`Button Create Button with {"text": "Hello World", "variant": "success"} 1`] = `
.emotion-0 {
  background-color: #3d899b;
  padding: 0.8333333333333334em 1.6666666666666667em;
  border: none;
  outline: none;
  height: 2.8333333333333335em;
  min-width: 7.5em;
  border-radius: 1.25em;
  font-style: normal;
  font-weight: 400;
  font-weight: normal;
  font-size: 0.8333333333333334em;
  color: #fff;
}

<button
  class="emotion-0"
  type="button"
>
  Hello World
</button>
`;
